server/utils.js

```diff
@@ -14,23 +15,23 @@ const fs = {
 	writeFile: denodeify(_fs.writeFile),
 	rename: denodeify(_fs.rename),
 	realpath: denodeify(_fs.realpath),
-	stat: denodeify(_fs.stat),
+	stat: denodeify(_fs.stat)
 }
 
-var slugify = function(str) {
+const slugify = function(str) {
 	str = str.replace(/[ \t\n_]+/g, "_")
 	str = str.replace(/[^\w_]+/g, "")
 	return str.toLowerCase()
 }
 
 //Recursive promises, weeee....
-var generateId = function(collection) {
-	let id = shortid.generate();
-	return collection.findOne({guid: id}).then((data) => data ? generateId(db) : id);
+const generateId = function(collection) {
+	const id = shortid.generate()
+	return collection.findOne({ guid: id }).then((data) => (data ? generateId(db) : id))
```

Comment:
I think that should be `generateId(collection)`.  If we've generated an ID that's already in this collection, generate a new one for this collection.